src/rust/ide/lib/view/src/graph_editor/component/project_name.rs

```diff
@@ -0,0 +1,314 @@
+//! This module provides a view for project's name which can be used to edit it.
+
+use crate::prelude::*;
+
+use enso_frp as frp;
+use ensogl::data::color;
+use ensogl::display;
+use ensogl::display::Attribute;
+use ensogl::display::Buffer;
+use ensogl::display::object::ObjectOps;
+use ensogl::display::scene::Scene;
+use ensogl::display::shape::text::text_field::TextField;
+use ensogl::display::shape::text::text_field::TextFieldProperties;
+use ensogl::display::shape::*;
+use ensogl::display::Sprite;
+use ensogl::display::world::World;
+use ensogl::gui::component::Animation;
+use ensogl::gui::component;
+use logger::enabled::Logger;
+use logger::AnyLogger;
+use nalgebra::Vector2;
+use ensogl::animation::linear_interpolation;
+
+
+
+// =================
+// === Constants ===
+// =================
+
+const HIGHLIGHTED_TEXT_COLOR : color::Rgba = color::Rgba::new(1.0,1.0,1.0,1.0);
+const DARK_GRAY_TEXT_COLOR   : color::Rgba = color::Rgba::new(1.0,1.0,1.0,0.6);
+
+/// Project name used as a placeholder in ProjectName view.
+pub const UNKNOWN_PROJECT_NAME:&str = "[UNKNOWN NAME]";
```

Comment:
docs should tell when it is used in the gui and in which situations it can appear.

Reply:
Moreover, it looks hacky, but I cannot tell anything how to fix it until thedocs tell when it is used.

Reply:
The doc now says:
`/// Project name used as a placeholder in ProjectName view when it's initialized.`

Please check this discussion: https://github.com/enso-org/ide/pull/639#discussion_r451362250

Reply:
Ok, it looks hacky, just like internal details showing to the user. Please change it to "Unknown". Is it ok @mwu-tow ?